lnst/Recipes/ENRT/ConfigMixins/DevFlowsPinningHWConfigMixin.py

```diff
@@ -0,0 +1,241 @@
+        """
+        Returns a value based on which should the flow be steered.
+
+        E.g. if :attr:`flow.steer_by` is "src-ip", it returns
+        :attr:`flow.generator_bind` IP address.
+
+        :param flow: flow object to pin
+        :returns: steering key
+        """
+        if steer_by == "src-ip":
+            return flow.generator_bind
+        elif steer_by == "dst-ip":
+            return flow.receiver_bind
+        elif steer_by == "src-port":
+            return flow.generator_port
+        elif steer_by == "dst-port":
+            return flow.receiver_port
+        else:
+            raise NotImplementedError(
+                f"Flow steering by {steer_by} not implemented (yet)."
+            )
+
+    def _get_flow_type(self, flow: Flow, steer_by) -> str:
+        """
+        Returns flow protocol based on which steering is done.
+        E.g.
+            if steering by IP, it returns "ip4" or "ip6".
+            if steering by TCP, it returns "tcp4" or "tcp6".
+
+        :param flow: flow object to pin
+        :returns: flow protocol+IP version
+        """
+        ip_version = (
+            "4" if isinstance(ipaddress(flow.receiver_bind), Ip4Address) else "6"
+        )
+
+        protocol = None
+        if "ip" in steer_by:  # e.g. src-ip, dst-ip
+            protocol = "ip"
+
+        elif "port" in steer_by:  # e.g. src-port, dst-port
+            if "tcp" in flow.type:
+                protocol = "tcp"
+            elif "sctp" in flow.type:
+                protocol = "sctp"
+            elif "udp" in flow.type or "xdp" in flow.type:  # xdp tests uses udp
+                protocol = "udp"
+            else:
+                raise NotImplementedError(
+                    f"Flow steering for {flow.type} not implemented (yet)."
+                )
+        else:
+            raise NotImplementedError(
+                f"Flow steering by {steer_by} not implemented (yet)."
+            )
+
+        return protocol + ip_version
+
+    def hw_deconfig(self, config):
+        if "flow_steering_rules" in config.hw_config:
+            for nic, rules in config.hw_config["flow_steering_rules"].items():
+                for id, _ in rules:
+                    nic.netns.run(f"ethtool -N {nic.name} delete {id}")
+                    # ^ remove steering rule
+
+        super().hw_deconfig(config)
+
+    def steer_flow_by(
+        self, flow
+    ) -> Optional[Literal["dst-ip", "src-ip", "dst-port", "src-port"]]:
+        """
+        This method should return an steering key, based on
+        which the flow should be steered across the queues.
+        E.g. if steering by src-ip, it returns "src-ip" and
+        the flow will be steered based on its' source IP address.
+
+        If None is returned, flow won't be steered.
+
+        :param flow: flow object to pin
+        :returns: steering key
+        """
+        raise NotImplementedError("Should be implemented in derived class.")
```

Comment:
and this is implemented where? i don't see it in this PR or any other PRs from you either?

Reply:
right but... that means this specific method with `NotImplementedError` is implemented somewhere, but not here...

Reply:
my point is... you're adding a config mixin class... which isn't used anywhere, not in this PR... or any other PR I see from you. so why do we need it?

here you have an example, but is this something we plan on running or including somewhere? what's the actual use of the mixin class?